activerecord/lib/active_record/autosave_association.rb

```diff
@@ -371,6 +372,11 @@ def before_save_collection_association
         true
       end
 
+      def after_save_collection_association
+        @new_record_before_save = false
+        true
```

Comment:
See, for example: https://github.com/rails/rails/blob/v5.1.0.rc1/activerecord/lib/active_record/autosave_association.rb#L220